src/clib/pio_file.c

```diff
@@ -117,7 +117,7 @@ int PIOc_createfile(const int iosysid, int *ncidp, int *iotype,
     file->buffer.data = NULL;
     file->buffer.next = NULL;
     file->buffer.vid = NULL;
-    file->buffer.ioid = -1;
+    file->buffer.ioid = 0;
```

Comment:
I think that there was a reason that this was -1 and not 0 - have you tested this change?